nwbwidgets/timeseries.py

```diff
@@ -423,10 +423,15 @@ def _prep_timeseries(time_series: TimeSeries, time_window=None, order=None):
         t_ind_stop = timeseries_time_to_ind(time_series, time_window[1])
 
     tt = get_timeseries_tt(time_series, t_ind_start, t_ind_stop)
-    unique_sorted_order, inverse_sort = np.unique(order, return_inverse=True)
 
     if len(time_series.data.shape) > 1:
-        mini_data = time_series.data[t_ind_start:t_ind_stop, unique_sorted_order][:, inverse_sort]
+        unique_sorted_order, inverse_sort = np.unique(order, return_inverse=True)
+        # fancy indexing is not supported in zarr, so we use slice when possible
+        if np.all(np.diff(unique_sorted_order) == 1):
+            unique_sorted_order = slice(unique_sorted_order[0], unique_sorted_order[-1] + 1)
+            mini_data = time_series.data[t_ind_start:t_ind_stop, unique_sorted_order][:, inverse_sort]
+        else:
+            mini_data = np.array(time_series.data[t_ind_start:t_ind_stop])[:, unique_sorted_order][:, inverse_sort]
```

Comment:
can we add a check here and have this work differently only for Zarr dataset objects? I'd prefer to use the simultaneous indexing approach for h5py datasets where we can so we don't load data into memory when we don't need to. I also think this could and probably should be refactored into a data utility that can be used in other places